pkg/operator/operator.go

```diff
@@ -475,9 +475,18 @@ func (o *Operator) sync(key string) error {
 		klog.Errorf("error occurred while setting status to in progress: %v", err)
 	}
 
-	taskName, err := tl.RunAll()
-	if err != nil {
-		o.reportError(err, taskName)
+	taskErrors := tl.RunAll()
+	if len(taskErrors) > 0 {
+		err := errors.New("Cluster monitoring update failed.")
```

Comment:
we don't need `err` anymore?

Reply:
This is the error that is returned from this function, I think it is required.